controller/src/main/java/org/jboss/as/controller/operations/common/ProcessEnvironment.java

```diff
@@ -117,6 +130,34 @@ protected abstract boolean isRuntimeSystemPropertyUpdateAllowed(String propertyN
      */
     protected abstract void systemPropertyUpdated(String propertyName, String propertyValue);
 
+    /**
+     * Get the UUID of this process.
+     * @return the UUID of this process.
+     */
+    public abstract UUID getInstanceUuid();
+
+    protected UUID obtainProcessUUID(final Path filePath) throws Exception {
+        UUID uuid = null;
+        if (Files.exists(filePath)) {
+            try (Stream<String> lines = Files.lines(filePath)) {
+                uuid = UUID.fromString(lines.findFirst().get());
+            }
+        }
+        if (uuid == null) {
+            PrivilegedExceptionAction<UUID> action = new PrivilegedExceptionAction<UUID>() {
```

Comment:
I was wrong about the needing of a privileged block. I should have looked more closely at where this was used. Sorry about that.